Chai `lengthOf` matcher not compatible with ethers/waffle objects

```
let quorum = ethers.constants.One;
let signers = await ethers.getSigners();

expect(quorum).to.be.at.least(1);

expect(signers.length).to.be.at.least(quorum);
expect(signers).to.have.lengthOf.at.least(1);
expect([1]).to.have.lengthOf.at.least(quorum);
expect(signers).to.have.lengthOf.at.least(quorum);
```

All statements run successfully except for the last one.

The error message is not particularly helpful: `AssertionError: Expected "[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" to be at least 1`

Previously reported to [Hardhat](https://github.com/nomiclabs/hardhat/issues/1157).

--- a/waffle-chai/src/matchers/bigNumber.ts
+++ b/waffle-chai/src/matchers/bigNumber.ts
@@ -42,6 +42,10 @@ function overwriteBigNumberFunction(
   return function (this: Chai.AssertionStatic, ...args: any[]) {
     const [actual] = args;
     const expected = chaiUtils.flag(this, 'object');
+    if (chaiUtils.flag(this, 'doLength') && BigNumber.isBigNumber(actual)) {
+      _super.apply(this, [actual.toNumber()]);
+      return;
+    }
     if (BigNumber.isBigNumber(expected) || BigNumber.isBigNumber(actual)) {
       this.assert(
         BigNumber.from(expected)[functionName](actual),

--- a/waffle-chai/test/matchers/misc.test.ts
+++ b/waffle-chai/test/matchers/misc.test.ts
@@ -1,4 +1,5 @@
 import {expect, AssertionError} from 'chai';
+import {ethers} from 'ethers';
 
 describe('UNIT: Miscellaneous', () => {
   describe('Proper address', () => {
@@ -78,6 +79,22 @@ describe('UNIT: Miscellaneous', () => {
       expect(
         () => expect('0x70').not.to.be.properHex(2)
       ).to.throw(AssertionError, 'Expected "0x70" not to be a proper hex of length 2, but it was');
+    });
+  });
+
+  describe('Other chai matchers compatibility', () => {
+    it('lengthOf works', async () => {
+      const one = ethers.constants.One;
+      const list = [1, 2, 3];
+
+      expect(one).to.be.at.least(1);
+
+      expect(list.length).to.be.at.least(one);
+      expect(list).to.have.lengthOf.at.least(1);
+      expect([1]).to.have.lengthOf.at.least(one);
+      expect(list).to.have.lengthOf.at.least(one);
+      expect(list).to.have.lengthOf.at.most(one.mul(3));
+      expect(() => expect(list).to.have.lengthOf.at.most(one)).to.throw;
     });
   });
 });

